samples/apps/research-assistant-react/backend/main.py

```diff
@@ -0,0 +1,690 @@
+            # Connect to the database (or create a new one if it doesn't exist)
+            conn = sqlite3.connect("database.sqlite")
+
+            # Create a cursor object to execute SQL commands
+            cursor = conn.cursor()
+
+            # Select the highest msgId from the messages table for the given userId and rootMsgId
+            cursor.execute(
+                "SELECT MAX(msgId) FROM messages WHERE userId = ? AND rootMsgId = ?",
+                (user_id, root_msg_id),
+            )
+
+            # Fetch the result
+            result = cursor.fetchone()
+
+            cursor.close()
+            # Close the connection
+            conn.close()
+
+            # Return the highest msgId if it exists, otherwise return -1
+            if result[0] is not None:
+                return result[0]
+            else:
+                return -1
+        finally:
+            lock.release()
+    except Exception as e:
+        print(f"Error getting highest msg_id from table table: {e}")
+
+
+def get_md5_hash(string):
+    import hashlib
+
+    return hashlib.md5(string.encode()).hexdigest()
+
+
+# Example usage:
+path_to_config_list = "./OAI_CONFIG_LIST"
+resulting_config = create_llm_config(path_to_config_list)
+
+
+def handle_message(message: Message, ra_type=AgentWorkFlow.CODER_ONLY, enable_personalization=False):
+    all_user_messages = get_messages(message.userId)["data"]
+    messages = all_user_messages[:-1]
+    # Inserting this assert because previous statement assumes that the last message
+    # in the database is from the user.
+    assert all_user_messages[-1]["role"] == "user"
+
+    # Setup the work_dir and utils_dir
+    usermd5 = get_md5_hash(message.userId)
+    global_utils_dir = get_global_utils_dir()
+    user_utils_dir = get_user_utils_dir(message.userId)
+    if not os.path.exists(user_utils_dir):
+        os.makedirs(user_utils_dir)
+    utils_dir = [global_utils_dir, user_utils_dir]
+
+    work_dir = os.path.join("files/user", usermd5, "work_dir")
+    if not os.path.exists(work_dir):
+        os.makedirs(work_dir)
+
+    # Get the user's personalization profile
+    personalization_profile = None
+    if enable_personalization:
+        personalization_profile = query_db(
+            "SELECT profile FROM personalization_profiles WHERE userId = ?",
+            (message.userId,),
+        )
+        if len(personalization_profile) > 0:
+            personalization_profile = personalization_profile[0]["profile"]
+
+    rprint(
+        """\n\n\n\n
+[italic red]--------------------------Handing new user message------------------------------[/italic red]
+\n\n\n\n
+"""
+    )
+    response = message_handler.process_user_message(
+        message.content.strip(),
+        history=messages,
+        utils_dir=utils_dir,
+        work_dir=work_dir,
+        ra_type=ra_type,
+        silent=False,
+        # agent_on_receive=pretty_print_agent_message,
+        path_to_config_list="./OAI_CONFIG_LIST",
```

Comment:
Hardcoded path. Inconsistent with instructions in the README.

Reply:
This is a quirk of autogen's `config_list_from_json`. It actually doesn't matter what this file path is.  If there's an environment variable, Autogen will use it.

Reply:
README asks to set a env variable but that isn't used here.

Reply:
Yeah, I mean we can remove this probably. My point was that if there's an OAI_CONFIG_LIST in os.environ, it won't even look at the filesystem. Having it in the environment supersedes everything else, so this is likely superfluous.

Reply:
Yes this is another reason consistency with AutoGen repo is important.